### DISCUSSION TITLE
Getting incorrect macos version (macos 10.16) using Python where runs-on set to macos-latest (11.6.8)

### DISCUSSION THREAD

**Discussion by swetayadav1** [actions/virtual-environments#6099] [General questions] [answered]
why python returns incorrect macos version 10.16 instead of 11.6.8?

**Image Release:** https://github.com/actions/virtual-environments/releases/tag/macOS-11%2F20220807.1

`Run python -c "import platform; print(platform.mac_ver()); print(platform.platform()); print(platform.uname()); print(open('/System/Library/CoreServices/SystemVersion.plist').read())"
  
('10.1[6](https://github.com/nuxeo/nuxeo-drive/runs/7860104243?check_suite_focus=true#step:7:6)', ('', '', ''), 'x8[6](https://github.com/nuxeo/nuxeo-drive/runs/7860104243?check_suite_focus=true#step:7:7)_64')
macOS-10.16-x86_64-i386-64bit
uname_result(system='Darwin', node='Mac-1660659681846.local', release='20.6.0', version='Darwin Kernel Version 20.6.0: Tue Jun 21 20:50:28 PDT 2022; root:xnu-[7](https://github.com/nuxeo/nuxeo-drive/runs/7860104243?check_suite_focus=true#step:7:8)195.141.32~1/RELEASE_X[8](https://github.com/nuxeo/nuxeo-drive/runs/7860104243?check_suite_focus=true#step:7:9)6_64', machine='x86_64')
<?xml version="1.0" encoding="UTF-8"?>`

**Comment by swetayadav1** [comment, marked as answer]
Any suggestion on this?

**Reply by al-cheb** [reply]
https://stackoverflow.com/questions/65290242/pythons-platform-mac-ver-reports-incorrect-macos-version - you can get more info about that behavior

```
- run: |
        SYSTEM_VERSION_COMPAT=0 python -c "import platform; print(platform.mac_ver()); print(platform.platform()); print(platform.uname()); print(open('/System/Library/CoreServices/SystemVersion.plist').read())"
```

![image](https://user-images.githubusercontent.com/47745270/185403608-8d75be0c-eb6f-4bb4-9219-fedf90de6a96.png)

**Comment by swetayadav1** [comment]
Thank you @al-cheb for your help. Used above approach only and it worked. 

I was using tox command ( run: tox -e unit)to run the test cases and there it was unable to detect SYSTEM_VERSION_COMPAT variable. After lot of search got to know, we need to set the reference in tox.ini file as well and it worked.

reference- https://tox.wiki/en/latest/config.html
we can use setenv (directly set key value) or "passenv" (Refer the env variable name defined in workflow)

#####################

[testenv]
setenv   =
    SYSTEM_VERSION_COMPAT = 0

OR

[testenv]
passenv = 
  SYSTEM_VERSION_COMPAT

############################

**Reply by al-cheb** [reply]
You can define an environment variable per step - https://docs.github.com/en/actions/learn-github-actions/environment-variables

```
     - name: "Run tests"
        run: echo "SYSTEM_VERSION_COMPAT = $SYSTEM_VERSION_COMPAT"
        env:
          SYSTEM_VERSION_COMPAT: 0
```

![image](https://user-images.githubusercontent.com/47745270/185580500-83c668a6-b9b7-4f2e-8e5d-7dcb0c72f5df.png)
